### ISSUE TITLE
Large taxonomy trees take longer to load than current BugGuide

### ISSUE THREAD

**Issue by kleintom** [bugguide/bugguide#440]
If you visit - warning: resource intensive link - https://beta.bugguide.net/node/3/tree?expand, chrome (96, on linux) seems to take ~90 seconds on my (older) computer. Firefox takes ~18 seconds, about the same as it takes to load https://bugguide.net/node/view/3/tree/all; chrome is slightly slower on the current version at ~23 seconds (these are all very rough times).

Offhand the main difference between beta and current that I'm aware of is that beta does tree indenting using `&nbsp;` while current does tree indenting using nested css (iirc). I guess we need to try the latter on beta.

**Comment by kleintom** [comment]
Ah, no need to guess, the network tab of the Developer Tools will tell you load times:

https://beta.bugguide.net/node/3/tree?expand --> 13.4 MB
on Chrome, 50.26 second load time - I've had load times on a hard refresh of anywhere from 45 seconds to 2.6 minutes
![image](https://user-images.githubusercontent.com/632915/146677718-cdfb96a3-6c0f-4f6e-a8f9-1edd4f38dd9a.png)
on Firefox, 19.27 seconds load time (consistently around 20 seconds)
![image](https://user-images.githubusercontent.com/632915/146665112-b10d4b56-409a-4a2b-805d-b03b7e646d9d.png)

https://bugguide.net/node/view/3/tree/all --> 12.4 MB
on Chrome, 23.02 seconds load time
![image](https://user-images.githubusercontent.com/632915/146665148-5deda9a2-4e21-4b5f-8cb8-2ba653c50d0e.png)
on Firefox, 12.55 seconds load time
![image](https://user-images.githubusercontent.com/632915/146665174-1aa9066c-121c-409a-9223-967f0e01cf74.png)

**Comment by kleintom** [comment]
Chrome is transferring content on the network nearly the entire time the page is loading (that's the blue line in the graph if I'm not mistaken), with a larger load up front and then smaller transfers (I think).
If I save the html of that page locally and load from the local file, I get consistent load times of ~50 seconds (with a mere ~350 milliseconds of file transfer time).

I just noticed though that if I hit the reload button and then immediately minimize the browser window, I get consistent load times of ~17 seconds on Chrome:
![image](https://user-images.githubusercontent.com/632915/146678741-b13cf66b-efac-4a16-aa4a-2f3a1e237f3e.png)
or ~8 seconds from the locally saved html file. Unfortunately that's of no use to us (or anyone else?) - is Chrome doing a lot of work trying to make sure content visible to the user is being kept up to date during initial load? If so why doesn't it have the same lag when loading the current BugGuide page, which isn't much smaller?

**Comment by kleintom** [comment]
One other difference I noticed between the current taxonomy layout and beta's taxonomy layout is that beta's is all one long line, while current has line breaks for each taxon item. I tried adding line breaks to the saved beta file, but that didn't make any difference in local load times (I think it made things slightly slower).

**Comment by clouseau** [comment]
Yes, this definitely a fun optimization problem. Which markup is the best for optimized indented rendering? Obviousness would say the smallest possible payload...but is that not true because some tags are parsed faster in the browser engine? Straight minimal HTML with tags that have indentation side effects vs CSS applied to more portly HTML? All of those approaches are probably faster than JS rendering of a huge nested JSON array. There is a lot to examine here!
